File: src/Z.Compression/ZipFile/System.IO.FileInfo/FileInfo.OpenReadZipFile.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
#if NET45_OR_GREATER
using System.IO;
using System.IO.Compression;

public static partial class Extensions
{
    /// <summary>
    ///     The path to the archive to open, specified as a relative or absolute path. A relative path is interpreted as
    ///     relative to the current working directory.
    /// </summary>
    /// <param name="this">
    ///     The path to the archive to open, specified as a relative or absolute path. A relative path is
    ///     interpreted as relative to the current working directory.
    /// </param>
    /// <returns>The opened zip archive.</returns>
    public static ZipArchive OpenReadZipFile(this FileInfo @this)
    {
        return ZipFile.OpenRead(@this.FullName);
    }
}
#endif

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Compression/ZipFile/System.IO.FileInfo/FileInfo.OpenReadZipFile.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

#If NET45_OR_GREATER Then
Imports System.IO
Imports System.IO.Compression

Public Module Extensions_50

	''' <summary>
	'''     The path to the archive to open, specified as a relative or absolute path. A relative path is interpreted as
	'''     relative to the current working directory.
	''' </summary>
	''' <param name="this">
	'''     The path to the archive to open, specified as a relative or absolute path. A relative path is
	'''     interpreted as relative to the current working directory.
	''' </param>
	''' <returns>The opened zip archive.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function OpenReadZipFile(this As FileInfo) As ZipArchive
		Return ZipFile.OpenRead(this.FullName)
	End Function
End Module
#End If


